<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>951</width>
    <height>553</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MGField</string>
  </property>
  <property name="iconSize">
   <size>
    <width>256</width>
    <height>256</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetMaximumSize</enum>
    </property>
    <item row="1" column="1">
     <widget class="QComboBox" name="CameraTypeComboBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>150</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="0" column="6">
     <widget class="Viewport3D" name="viewport3D">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="CameraTypeLabel">
      <property name="text">
       <string>Camera Type</string>
      </property>
     </widget>
    </item>
    <item row="1" column="6">
     <widget class="QPushButton" name="simulatePushButton">
      <property name="text">
       <string>Simulate</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="2">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QPushButton" name="AddObjectButton">
          <property name="text">
           <string>Add Object</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="RemoveObjectButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Remove Object</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="ObjectListView" name="ObjectList"/>
      </item>
     </layout>
    </item>
    <item row="0" column="8">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="ViewportSettings" name="ViewportSettingsWidget">
        <property name="title">
         <string>Viewport Settings</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ObjectInspector" name="ObjectInspectorInstance">
        <property name="title">
         <string>Object Inspector</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3"/>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>951</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuSimulation">
    <property name="title">
     <string>Simulation</string>
    </property>
    <addaction name="actionSimulationSettings"/>
    <addaction name="actionSimulationClearResults"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSimulation"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="font">
    <font>
     <family>JetBrains Mono</family>
    </font>
   </property>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionSimulationSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionSimulationClearResults">
   <property name="text">
    <string>Clear Results</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Viewport3D</class>
   <extends>QOpenGLWidget</extends>
   <header>src/view/viewport3d.h</header>
  </customwidget>
  <customwidget>
   <class>ObjectListView</class>
   <extends>QListView</extends>
   <header>src/view/objectlistview.h</header>
  </customwidget>
  <customwidget>
   <class>ViewportSettings</class>
   <extends>QGroupBox</extends>
   <header>src/view/viewportsettings.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ObjectInspector</class>
   <extends>QGroupBox</extends>
   <header>src/view/objectinspector.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
